internal/driver/macmapper.go

```diff
@@ -0,0 +1,113 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package driver
+
+import (
+	"fmt"
+	"net"
+	"strings"
+	"sync"
+
+	sdk "github.com/edgexfoundry/device-sdk-go/v2/pkg/service"
+)
+
+type MACAddressMapper struct {
+	// credsMu is for locking access to the credsMap
+	credsMu sync.RWMutex
+	// credsMap is a map between mac address to secretPath
+	credsMap map[string]string
+}
+
+// NewMACAddressMapper creates a new MACAddressMapper object
+func NewMACAddressMapper() *MACAddressMapper {
+	return &MACAddressMapper{
+		credsMap: make(map[string]string),
+	}
+}
+
+// UpdateMappings takes the raw map of secret path to csv list of mac addresses and
+// inverts it into a quick lookup map of mac address to secret path.
+func (m *MACAddressMapper) UpdateMappings(raw map[string]string) {
+	m.credsMu.Lock()
+	defer m.credsMu.Unlock()
+
+	lc := sdk.RunningService().LoggingClient
+
+	credsMap := make(map[string]string)
+	for secretPath, macs := range raw {
+		if _, err := sdk.RunningService().SecretProvider.GetSecret(secretPath, UsernameKey, PasswordKey, AuthModeKey); err != nil {
+			lc.Warnf("One or more MAC address mappings exist for the secret path '%s' which does not exist in the Secret Store!", secretPath)
+		}
+
+		for _, mac := range strings.Split(macs, ",") {
+			sanitized, err := SanitizeMACAddress(mac)
+			if err != nil {
+				lc.Warnf("Skipping invalid mac address %s: %s", mac, err.Error())
+				continue
+			}
+			// note: if the mac address already has a mapping, we do not overwrite it
+			if existing, found := credsMap[sanitized]; found {
+				lc.Warnf("Unable to set credential group to %s. MAC address '%s' already belongs to credential group %s.", secretPath, mac, existing)
+			} else {
+				credsMap[sanitized] = secretPath
+			}
+		}
+	}
+
+	m.credsMap = credsMap
+}
+
+// ListMACAddresses will return a slice of mac addresses that have been assigned credentials
+func (m *MACAddressMapper) ListMACAddresses() []string {
```

Comment:
Note that this was part of checking off the acceptence criteria, but the function isn't called, so not sure if we even need it.

Reply:
Todd clarified it will be needed for the next story 7236